### PR TITLE
WIP: Modify MRReply_ToValue to handle double conversion

### PR DESCRIPTION
**Pull request by nafraf**
## Describe the changes in the pull request

A clear and concise description of what the PR is solving, including:
1. Current: The current state briefly
2. Change: What is the change
3. Outcome: Adding the outcome

#### Which additional issues this PR fixes
1. MOD-...
2. #...

#### Main objects this PR modified
1. ...

#### Mark if applicable

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/coord/dist_aggregate.c [hunk 1]
```diff
@@ -165,14 +165,25 @@ static void netCursorCallback(MRIteratorCallbackCtx *ctx, MRReply *rep) {
...
       const char *s = MRReply_String(r, &l);
       v = RS_NewCopiedString(s, l);
       break;
     }
+    case MR_REPLY_STRING: {
+      double d = 0;
+      // Try to convert to double because RedisModule_Reply_Double() uses
+      // RedisModule_ReplyWithDouble() which converts double into string
+      if (MRReply_ToDouble(r, &d)) {
+        v = RS_NumVal(d);
```
Comment (oshadmi) [inline src/coord/dist_aggregate.c:180]: So it related to RESP2 "limitation"
https://github.com/redis/redis/blob/74b289a0e12f9f65a6daeec6a66cadc76792f644/src/module.c#L3481C2-L3482C3
```
 * In RESP3 the string is tagged as a double, while in RESP2 it's just a plain string 
 * that the user will have to parse.
```

File: src/value.c [hunk 2]
```diff
@@ -670,15 +670,11 @@ int RSValue_SendReply(RedisModule_Reply *reply, const RSValue *v, SendReplyFlags
...
+        // With STRING format and with `_NUM_SSTRING` disabled, return the number as a string
         char buf[128];
         size_t len = RSValue_NumToString(v->numval, buf);
-
-        if (flags & SENDREPLY_FLAG_TYPED) {
-          return RedisModule_Reply_Error(reply, buf);
-        } else {
-          return RedisModule_Reply_StringBuffer(reply, buf, len);
-        }
+        return RedisModule_Reply_StringBuffer(reply, buf, len);
```
Comment (oshadmi) [inline src/value.c:677]: Why not use existing PR https://github.com/RediSearch/RediSearch/pull/5187

Comment (nafraf) [reply src/value.c:677]: OK. I'll include this changes in the existing PR.

**Comment by nafraf** [comment]
Closing this PR because these changes were included in #5187

**Pull request closed without merging**